rfc/rfc-0000-maven-artifacts-support.md

```diff
@@ -0,0 +1,130 @@
+# Meta
+[meta]: #meta
+- Name: Maven Artifacts Support
+- Start Date: 2022-03-30
+- Author(s): @emmjohnson
+- Status: Draft
+- RFC Pull Request: (leave blank)
+- Supersedes: N/A
+
+# Summary
+[summary]: #summary
+
+Modify `owner.spec.source` to allow for specifying a Maven and Artifactory repository.
+
+# Motivation
+[motivation]: #motivation
+
+- Why should we do this?
+  - Today we support workloads being built from git and image source. Many developers in the field also need support
+    Maven and Artifactory repositories hosting their JAR and WAR files.
+- What use cases does it support?
+  - Maven and Artifactory repositories
+- What is the expected outcome?
+  - To enable more types of workloads.
```

Comment:
@emmjohnson should the reference to `Artifactory` be more specific and not confused as a source type here? Similar to the comment in https://github.com/vmware-tanzu/cartographer/pull/767#discussion_r839942652